editors/code/src/config.ts

```diff
@@ -149,8 +154,12 @@ export class Config {
     get highlightingOn() { return this.cfg.get("highlightingOn") as boolean; }
     get rainbowHighlightingOn() { return this.cfg.get("rainbowHighlightingOn") as boolean; }
     get lruCapacity() { return this.cfg.get("lruCapacity") as null | number; }
-    get displayInlayHints() { return this.cfg.get("displayInlayHints") as boolean; }
-    get maxInlayHintLength() { return this.cfg.get("maxInlayHintLength") as number; }
+    get inlayHintOpts(): InlayHintOptions {
+        return {
+            displayType: this.cfg.get("inlayHintOpts.displayType") as string,
```

Comment:
If I get it right, changing both inlay hint properties require the server reload now.
It would be super cool, if you could add thoses properties into the `requiresReloadOpts` so a corresponding warning message pops up.

Reply:
@SomeoneToIgnore, @kjeremy  I guess in order to not require any reload we need to move configurations from `initializationOptions` to dynamic config and handle it somewhere here in `DidChangeConfiguration` notification, right?:
https://github.com/rust-analyzer/rust-analyzer/blob/0dbd8ff59b854570329c325431126a078505e5f5/crates/rust-analyzer/src/main_loop.rs#L618-L623

Reply:
@Veetaha correct. Think of `initializationOptions` as program arguments. For dynamic configuration see: https://github.com/microsoft/language-server-protocol/issues/676 on how `workspace/didChangeConfiguration` and `workspace/configuration` should interact.

Reply:
@SomeoneToIgnore, also, you know, Github suggestions feature is super cool for this kind of collaboration (especially when they've added an ability to create multiline suggestions!)

Reply:
Should I add the properties to the `requiresReloadOpts` for now and do a separate PR for loading them dynamically.

Reply:
Yeah, `requiresReloadOpts` is what we live for now. Adding dynamic config reloading will touch quite a lot of config values, not only inlay hints, so that is an unrelated separate feature.